demonstrations/tutorial_rosalin.py

```diff
@@ -436,7 +447,12 @@ def estimate_hamiltonian(self, params, shots):
         shots_per_term = si.rvs()[0]
 
         results = []
-        for h, c, p, s in zip(self.qnodes, self.coeffs, prob_shots, shots_per_term):
+        for o, c, p, s in zip(self.obs, self.coeffs, prob_shots, shots_per_term):
+
+            @qml.qnode(rosalin_device, diff_method="parameter-shift")
+            def h(weights):
+                StronglyEntanglingLayers(weights, wires=rosalin_device.wires)
+                return qml.sample(o)

```

Comment:
```suggestion
        @qml.qnode(rosalin_device, diff_method="parameter-shift")
        def h(weights, obs):
            StronglyEntanglingLayers(weights, wires=rosalin_device.wires)
            return qml.sample(obs)

        for o, c, p, s in zip(self.obs, self.coeffs, prob_shots, shots_per_term):

```

And update the code inside the loop to pass in the observable to `h` accordingly

Reply:
Thanks for noticing this!